### ISSUE TITLE
'Input string was not in a correct format' when clicking on some areas of chart tree

### ISSUE THREAD

**Issue by Lafil** [ITLec/ITLec.CRMChartGuy.XRMToolBox.AdvancedChartEditor#1]
Using version 1.2017.6.11
When navigating through part of the chart tree in the visual editor, you can get the following error:
'Input string was not in a correct format'.
This happens on both custom charts and OOB charts.
For example:
Used on-premises org with OOB chart "Case Mix (By Origin)" on Case (incident) entity.
You can get the error by clicking on the Chart / Series / Series area.

**Comment by ITLec** [comment]
I have tried this scenario, and it is working fine for me.
Please navigate to "%APPDATA%\MscrmTools\XrmToolBox\Settings\ChartStructure.xml", and send to me the content of this file.
also send to me the content of the XML in Chart Editor Screen > Presentation Desc,  
please find this image http://i.imgur.com/WfAx8KV.png

**Comment by Lafil** [comment]
Hi Mohamed,

Here is the chartstructure.xml file

Here is the content of the presentation description:
<Chart Palette="None" PaletteCustomColors="91,151,213; 237,125,49; 160,116,166; 255,192,0; 68,114,196; 112,173,71; 37,94,145; 158,72,14; 117,55,125; 153,115,0; 38,68,120; 67,104,43; 124,175,221; 241,151,90; 186,144,192; 255,205,51; 105,142,208; 140,193,104; 50,125,194; 210,96,18; 150,83,159; 204,154,0; 51,90,161; 90,138,57;">
  <Series>
    <Series ShadowOffset="0" IsValueShownAsLabel="true" Font="{0}, 9.5px" LabelForeColor="59, 59, 59" CustomProperties="PieLabelStyle=Inside, PieDrawingStyle=Default" ChartType="pie">
      <SmartLabelStyle Enabled="True" />
    </Series>
  </Series>
  <ChartAreas>
    <ChartArea>
      <Area3DStyle Enable3D="false" />
    </ChartArea>
  </ChartAreas>
  <Legends>
    <Legend Alignment="Center" LegendStyle="Table" Docking="right" Font="{0}, 11px" ShadowColor="0, 0, 0, 0" ForeColor="59, 59, 59" />
  </Legends>
  <Titles>
    <Title Alignment="TopLeft" DockingOffset="-3" Font="{0}, 13px" ForeColor="0, 0, 0">
    </Title>
  </Titles>
</Chart>

Thanks,
L-A


De : Mohamed Rasheed Gomaa (ITLec) [mailto:notifications@github.com]
Envoyé : 12 juin 2017 21:46
À : ITLec/ITLec.CRMChartGuy.XRMToolBox.AdvancedChartEditor <ITLec.CRMChartGuy.XRMToolBox.AdvancedChartEditor@noreply.github.com>
Cc : Lafil <lafil@hotmail.com>; Author <author@noreply.github.com>
Objet : Re: [ITLec/ITLec.CRMChartGuy.XRMToolBox.AdvancedChartEditor] 'Input string was not in a correct format' when clicking on some areas of chart tree (#1)


I have tried this scenario, and it is working fine for me.
Please navigate to "%APPDATA%\MscrmTools\XrmToolBox\Settings\ChartStructure.xml", and send to me the content of this file.
also send to me the content of the XML in Chart Editor Screen > Presentation Desc, http://i.imgur.com/WfAx8KV.png<url>

**Comment by ITLec** [comment]
Hi Lafil ,

First of all, I would like to thank you for your time and reporting an issue to us about AdvancedChartEditor tool.

I have fixed the reported issue, I need your confirmation so I can update the tool in the xrmtoolbox store.

To test the updated plugin:-
1- Navigate to "%APPDATA%\MscrmTools\XrmToolBox\Plugins\".
2- Replace the "ITLecChartGuy.AdvancedChartEditor.dll", with the attached dll 
 https://www.4shared.com/zip/HNfqqEwAca/ITLecChartGuyAdvancedChartEdit.html?  (You must extract the attached file first to get the dll file).
Note: In case the update is working fine, please mark this issue as solved.
Thanks,
Rasheed

**Comment by Lafil** [comment]
Hi Rasheed,

Thanks for the quick follow up on the issue!
Unfortunately, 4shared is blocked on our network.
I have created a folder on my OneDrive where you could drop the updated dll if that’s ok?
https://1drv.ms/f/s!AjttlZ635-1WmAs7BzBboiY_RNwp

Thanks,
Louis-Alain

De : Mohamed Rasheed Gomaa (ITLec) [mailto:notifications@github.com]
Envoyé : 13 juin 2017 22:16
À : ITLec/ITLec.CRMChartGuy.XRMToolBox.AdvancedChartEditor <ITLec.CRMChartGuy.XRMToolBox.AdvancedChartEditor@noreply.github.com>
Cc : Lafil <lafil@hotmail.com>; Author <author@noreply.github.com>
Objet : Re: [ITLec/ITLec.CRMChartGuy.XRMToolBox.AdvancedChartEditor] 'Input string was not in a correct format' when clicking on some areas of chart tree (#1)


Hi Lafil ,

First of all, I would like to thank you for your time and reporting an issue to us about AdvancedChartEditor tool.

I have fixed the reported issue, I need your confirmation so I can update the tool in the xrmtoolbox store.

To test the updated plugin:-
1- Navigate to "%APPDATA%\MscrmTools\XrmToolBox\Plugins".
2- Replace the "ITLecChartGuy.AdvancedChartEditor.dll", with the attached dll
https://www.4shared.com/zip/HNfqqEwAca/ITLecChartGuyAdvancedChartEdit.html? (You must extract the attached file first to get the dll file).
Note: In case the update is working fine, please mark this issue as solved.
Thanks,
Rasheed

**Comment by ITLec** [comment]
Hi Louis,
I have uploaded in the mentioned url.

**Comment by Lafil** [comment]
Hi Rasheed,

Got it, thank you!

It resolved the problem with the example I submitted.
I tried various OOB charts and it seems to navigate correctly.
However, I am getting an error on one of my custom case entity based charts. See attached XML.
It runs off case entity’s OOB attributes so you should be ok to import it if you need to.
The plugin gives an error when trying to go to the Chart / Series / Series / CustomProperties node.
“Object reference not set to an instance of an object”.

Would you like me to submit a separate issue for this, or do you believe it is a variation of the same issue?

Best,
Louis-Alain

[cid:image001.png@01D2E52A.0127FA30]




De : Mohamed Rasheed Gomaa (ITLec) [mailto:notifications@github.com]
Envoyé : 14 juin 2017 15:18
À : ITLec/ITLec.CRMChartGuy.XRMToolBox.AdvancedChartEditor <ITLec.CRMChartGuy.XRMToolBox.AdvancedChartEditor@noreply.github.com>
Cc : Lafil <lafil@hotmail.com>; Author <author@noreply.github.com>
Objet : Re: [ITLec/ITLec.CRMChartGuy.XRMToolBox.AdvancedChartEditor] 'Input string was not in a correct format' when clicking on some areas of chart tree (#1)


Hi Louis,
I have uploaded in the mentioned url.

**Comment by ITLec** [comment]
Hi Louis,

I am happy to hear that "This issue has been solved".

and I am highly appreciate your contribution and figure out some issues.

I prefer to close this issue, and create another one as it is another issue.
